powerline_shell/segments/svn.py

```diff
@@ -1,29 +1,51 @@
 import subprocess
-from ..utils import BasicSegment, RepoStats
+from ..utils import ThreadedSegment, RepoStats
 
 
-class Segment(BasicSegment):
-    def add_to_powerline(self):
-        is_svn = subprocess.Popen(["svn", "status"],
-                                  stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        is_svn_output = is_svn.communicate()[1].decode("utf-8").strip()
-        if len(is_svn_output) != 0:
-            return
+class Segment(ThreadedSegment):
+    def __init__(self, powerline):
+        super().__init__(powerline)
+        self.stats = None
+        self.revision = ""
 
+    def run(self):
         try:
-            p1 = subprocess.Popen(["svn", "status"], stdout=subprocess.PIPE,
-                    stderr=subprocess.PIPE)
+            svn_status = subprocess.Popen(["svn", "status"],
+                                          stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            svn_info = subprocess.Popen(["svn", "info"],
+                                        stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            svn_stdout, svn_stderr = svn_status.communicate()
+            svn_info, _ = svn_info.communicate()
         except OSError:
             return
 
-        stdout = p1.communicate()[0]
-        stats = RepoStats()
-        for line in stdout.splitlines():
+        if len(svn_stderr.decode("utf-8").strip()) != 0:
+            return
+
+        self.stats = RepoStats()
+        for line in svn_stdout.splitlines():
+            line = line.decode("utf-8").strip()
             if line[0] == "?":
-                stats.new += 1
+                self.stats.new += 1
             elif line[0] == "C":
-                stats.conflicted += 1
+                self.stats.conflicted += 1
             elif line[0] in ["A", "D", "I", "M", "R", "!", "~"]:
-                stats.changed += 1
+                self.stats.changed += 1
+
+        for line in svn_info.splitlines():
+            line = line.decode("utf-8").strip()
+            if "Revision: " in line:
```

Comment:
There's an issue with locales other than English because `svn info`'s output is localized. I fixed it in commit 235e02b8312935122dc9e4ccc3c041f80583a8da, since I had to merge it anyway. My solution was to use `svn info --xml`.